jaq-core/src/lib.rs

```diff
@@ -402,3 +404,93 @@ const LOG: &[(&str, usize, RunPtr, UpdatePtr)] = &[(
     |_, cv| box_once(Ok(debug(cv.1))),
     |_, cv, f| f(debug(cv.1)),
 )];
+
+#[cfg(feature = "format")]
+fn to_csv(v: Val, delimiter: u8) -> ValR {
+    use csv::{StringRecord, WriterBuilder};
+    match v.clone() {
+        Val::Arr(vs) => {
+            let mut writer = WriterBuilder::new()
+                .delimiter(delimiter)
+                .from_writer(vec![]);
+            let mut record = StringRecord::new();
+            for cell in &*vs {
+                match cell {
+                    Val::Str(s) => {
+                        record.push_field(s);
+                        Ok(())
+                    }
+                    _ => Err(Error::Str(cell.clone())),
+                }?;
+            }
+            writer
+                .write_record(&record)
+                .map_err(|e| Error::Csv(e.to_string()))?;
+            Ok(Val::str(
+                String::from_utf8(writer.into_inner().map_err(|e| Error::Csv(e.to_string()))?)
+                    .map_or_else(
+                        |e| Err(Error::Csv(e.to_string())),
+                        |s| Ok(s[0..s.len() - 1].to_string()),
+                    )?,
+            ))
+        }
+        _ => Err(Error::Arr(v)),
+    }
+}
+
+#[cfg(feature = "format")]
+const FORMAT: &[(&str, usize, RunPtr)] = &[
+    ("tohtml", 0, |_, cv| {
+        box_once(match cv.1 {
+            Val::Str(s) => Ok(Val::str(html_escape::encode_safe(&*s).to_string())),
+            _ => Err(Error::Str(cv.1)),
+        })
+    }),
+    ("touri", 0, |_, cv| {
+        box_once(match cv.1 {
+            Val::Str(s) => Ok(Val::str(urlencoding::encode(&s).to_string())),
+            _ => Err(Error::Str(cv.1)),
+        })
```

Comment:
You frequently have this pattern where you check if the input value is a string, and if it is, then run some function on the string and return another string.
I would create a function in `Val` that encodes this pattern, similarly to `mutate_str()`. You could call it `map_str`, for example. For functions that can fail during this conversion, like `fromuri`, you could introduce another function like `try_map_str()`.

Reply:
This remark of mine is probably obsolete if you consider [my comment below](https://github.com/01mf02/jaq/pull/105#discussion_r1289774934).